In [1]:
import pandas as pd;

In [2]:
df=pd.read_csv("D:\Toyota.csv");

In [3]:
df=pd.read_csv("D:\Toyota.csv" ,index_col=0)
print(df.isnull().sum().sum())
print(df.shape)

350
(1436, 10)


In [4]:
# importing data 
cars_data=pd.read_csv("D:\Toyota.csv" ,index_col=0,na_values=["?","???"])

In [5]:
# Creating copy of original data 
cars_data2= cars_data.copy()


In [6]:
# Frequency table : you have the corresponding categories under the variable fueal type 
Frequency_table = pd.crosstab(index=cars_data2['FuelType'], columns='count',dropna=True )

In [7]:
print(Frequency_table)

col_0     count
FuelType       
CNG          15
Diesel      144
Petrol     1177


# From the above output :
 CNG: There are 15 cars that use CNG as their fuel type. This is a very small proportion of the dataset, indicating that CNG is not a common fuel type among the cars in this dataset.
 
#Diesel: There are 144 cars that use diesel as their fuel type. This represents a moderate proportion of the dataset, suggesting that diesel cars are somewhat common.

#Petrol: There are 1177 cars that use petrol as their fuel type. This is the majority fuel type in the dataset, indicating that petrol is the most common fuel type among the cars.
    

In [8]:
# Frequency distribution of gearbox types with respect to different fueal type of the car 
automatic_fuel_type_crosstab = pd.crosstab(
    index=cars_data2['Automatic'],
    columns=cars_data2['FuelType'],
    dropna=True
)

In [10]:
print(automatic_fuel_type_crosstab)

FuelType   CNG  Diesel  Petrol
Automatic                     
0           15     144    1104
1            0       0      73


# From the above output

 All CNG and diesel cars have manual transmissions, with no automatic options available.
 
Among petrol cars, a small percentage (73 out of 1177, or approximately 6.2%) have automatic transmissions.
Automatic transmission is available only for petrol cars, not for CNG or diesel cars in this dataset.

In [11]:
# Crosstab for 'Automatic' and 'FuelType' with normalization
automatic_fuel_type_crosstab_joint_Probabiliy  = pd.crosstab(
    index=cars_data2['Automatic'],
    columns=cars_data2['FuelType'],
    normalize=True,
    dropna=True
)

In [12]:
print(automatic_fuel_type_crosstab_joint_Probabiliy)

FuelType        CNG    Diesel    Petrol
Automatic                              
0          0.011228  0.107784  0.826347
1          0.000000  0.000000  0.054641


# Manual Cars (Automatic = 0):::
    1.1% are CNG, 10.8% are Diesel, and 82.6% are Petrol.

# Automatic Cars (Automatic = 1):::
    0% are CNG or Diesel, and 5.5% are Petrol.
    
# Overall, most cars with manual transmission are petrol (82.6%), and a small percentage (5.5%) of petrol cars have automatic transmission.








In [13]:
# Two way table  Marginal probability 
automatic_fuel_type_crosstab_Marginal_Probabiliy  = pd.crosstab(
    index=cars_data2['Automatic'],
    columns=cars_data2['FuelType'],
    margins=True,
    normalize=True,
    dropna=True
)


In [14]:
print(automatic_fuel_type_crosstab_Marginal_Probabiliy)

FuelType        CNG    Diesel    Petrol       All
Automatic                                        
0          0.011228  0.107784  0.826347  0.945359
1          0.000000  0.000000  0.054641  0.054641
All        0.011228  0.107784  0.880988  1.000000


# Manual Cars (Automatic = 0):
1.1% are CNG, 10.8% are Diesel, and 82.6% are Petrol, making up 94.5% of the total cars.

# Automatic Cars (Automatic = 1):
0% are CNG or Diesel, and 5.5% are Petrol, making up 5.5% of the total cars.

# Overall (All):
1.1% of all cars are CNG, 10.8% are Diesel, and 88.1% are Petrol.

Most cars are petrol (88.1%) and have manual transmission (94.5%). Only 5.5% of the cars have automatic transmission, and these are exclusively petrol.








In [15]:
# Two way table conditional probability
# Given the type of gear box ,probability of different fueal type
automatic_fuel_type_crosstab_Conditional_Probabiliy  = pd.crosstab(
    index=cars_data2['Automatic'],
    columns=cars_data2['FuelType'],
    margins=True,
    normalize='index',
    dropna=True
)

In [16]:
print(automatic_fuel_type_crosstab_Conditional_Probabiliy )

FuelType        CNG    Diesel    Petrol
Automatic                              
0          0.011876  0.114014  0.874109
1          0.000000  0.000000  1.000000
All        0.011228  0.107784  0.880988


# Manual Cars (Automatic = 0):
1.2% are CNG, 11.4% are Diesel, and 87.4% are Petrol.

# Automatic Cars (Automatic = 1):
0% are CNG or Diesel, and 100% are Petrol.

#Overall (All):
1.1% of all cars are CNG, 10.8% are Diesel, and 88.1% are Petrol.

This shows that all automatic cars are petrol, while manual cars have a mix of CNG, Diesel, and Petrol, with petrol being the most common fuel type in both manual and automatic cars.








In [17]:

# Here I am changing Normalize to Column just to get the column sums as 1
automatic_fuel_type_crosstab_Conditional__Probabiliy  = pd.crosstab(
    index=cars_data2['Automatic'],
    columns=cars_data2['FuelType'],
    margins=True,
    normalize='columns',
    dropna=True
)

In [18]:
print(automatic_fuel_type_crosstab_Conditional__Probabiliy)

FuelType   CNG  Diesel    Petrol       All
Automatic                                 
0          1.0     1.0  0.937978  0.945359
1          0.0     0.0  0.062022  0.054641


# CNG and Diesel Cars:
100% of CNG and Diesel cars are manual (Automatic = 0).

# Petrol Cars:
93.8% of Petrol cars are manual (Automatic = 0).
6.2% of Petrol cars are automatic (Automatic = 1).

#Overall (All):
94.5% of all cars are manual.
5.5% of all cars are automatic.

This indicates that CNG and Diesel cars are exclusively manual, while a small proportion of petrol cars are automatic. Most cars, regardless of fuel type, are manual.







